### PR TITLE
Close FileInputStream and FileOutputStream resources after usage

### PR DESCRIPTION
**Pull request by malshaniS**
## Automation tests
 - Integration tests

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA3336HostHeaderValuePortCheckTestCase.java [hunk 1]
```diff
@@ -93,9 +93,13 @@ public void stop() throws Exception {
...
-        properties.load(new FileInputStream(srcFile));
+        FileInputStream fis = new FileInputStream(srcFile);
+        properties.load(fis);
         properties.setProperty(key, value);
-        properties.store(new FileOutputStream(destinationFile), null);
+        fis.close();
+        FileOutputStream fos = new FileOutputStream(destinationFile);
+        properties.store(fos, null);
         serverConfigurationManager.applyConfigurationWithoutRestart(destinationFile);
+        fos.close();
```
Comment (ThishaniLucas) [inline integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA3336HostHeaderValuePortCheckTestCase.java]: It's better if we could handle this inside a try-catch-finally block.

Comment (malshaniS) [reply integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA3336HostHeaderValuePortCheckTestCase.java]: Fixed. Please refer latest commit

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA3336HostHeaderValuePortCheckTestCase.java [hunk 2]
```diff
@@ -91,11 +93,27 @@ public void stop() throws Exception {
...
+        try {
+            File destinationFile = new File(outFileName);
+            fis = new FileInputStream(srcFile);
+            fos = new FileOutputStream(destinationFile);
+            Properties properties = new Properties();
+            properties.load(fis);
+            properties.setProperty(key, value);
+            properties.store(fos, null);
+            serverConfigurationManager.applyConfigurationWithoutRestart(destinationFile);
+        } catch (Exception e) {
```
Comment (yasassri) [inline integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA3336HostHeaderValuePortCheckTestCase.java]: what if applyConfiguration fails or any other error occurs? Do you still want to run the test case?

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA3336HostHeaderValuePortCheckTestCase.java [hunk 3]
```diff
@@ -18,6 +18,7 @@
 
 package org.wso2.carbon.esb.passthru.transport.test;
 
+import junit.framework.Assert;
```
Comment (yasassri) [inline integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA3336HostHeaderValuePortCheckTestCase.java]: Should we use TestNG assertions? Is there a usage for this?

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA3336HostHeaderValuePortCheckTestCase.java (hunk 2 above)
Comment (malshaniS) [reply integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA3336HostHeaderValuePortCheckTestCase.java]: Fixed. Please refer latest commit

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA3336HostHeaderValuePortCheckTestCase.java (hunk 3 above)
Comment (malshaniS) [reply integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA3336HostHeaderValuePortCheckTestCase.java]: Removed the assertion. Please refer latest commit.

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/HttpAccessLogTestCase.java [hunk 4]
```diff
@@ -125,13 +125,29 @@ private void createNewDir(String path) throws IOException {
      * @param value
      * @throws Exception
```
Comment (yasassri) [inline integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/HttpAccessLogTestCase.java]: Can we fix the doc comment here?

Comment (malshaniS) [reply integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/HttpAccessLogTestCase.java]: Fixed the doc comment

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/HttpAccessLogTestCase.java [hunk 5]
```diff
@@ -123,15 +123,31 @@ private void createNewDir(String path) throws IOException {
...
+            if (fis != null) {
+                fis.close();
+            }
+            if (fos != null) {
+                fos.close();
+            }
+        }
     }
 
     /**
```
Comment (dilee) [inline integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/HttpAccessLogTestCase.java]: Remove this unused comment tag.

**Pull request merged**